x-pack/plugins/file_upload/server/routes.ts

```diff
@@ -37,7 +40,55 @@ function importData(
 /**
  * Routes for the file upload.
  */
-export function fileUploadRoutes(router: IRouter) {
+export function fileUploadRoutes(coreSetup: CoreSetup<StartDeps, unknown>, logger: Logger) {
+  const router = coreSetup.http.createRouter();
+
+  router.get(
+    {
+      path: '/internal/file_upload/has_import_permission',
+      validate: {
+        query: schema.object({
+          indexName: schema.maybe(schema.string()),
+          checkCreateIndexPattern: schema.boolean(),
+          hasPipeline: schema.boolean(),
+        }),
+      },
+    },
+    async (context, request, response) => {
+      try {
+        const [, pluginsStart] = await coreSetup.getStartServices();
+        const { indexName, checkCreateIndexPattern, hasPipeline } = request.query;
+
+        const authorizationService = pluginsStart.security?.authz;
+        const requiresAuthz = authorizationService?.mode.useRbacForRequest(request) ?? false;
+
+        if (!authorizationService || !requiresAuthz) {
```

Comment:
So we're defaulting to saying the user has permission has import permissions if `authorizationService` isn't available?

Reply:
That is correct. That is what security recommended to do if the security plugin is disabled. This is just a preflight check so any privileges check will still happen at the time of index creation and document creation